uk.co.compucorp.civicrm.tasksassignments/tests/phpunit/CRM/Tasksassignments/BAO/AssignmentTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+
+/**
+ *
+ * @group headless
+ */
+class CRM_Tasksassignments_BAO_AssignmentTest extends \PHPUnit_Framework_TestCase implements HeadlessInterface, TransactionalInterface {
+
+  use TasksassignmentsTestTrait;
+
+  private $contact_id;
+
+  private $case_type_id;
+
+  private $open_status_id;
+
+  private $closed_status_id;
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->installMe(__DIR__)
+      ->apply();
+  }
+
+  public function setUp() {
+    parent::setUp();
+    // Create Case Type
+    $params = ['first_name' => 'micky', 'last_name' => 'mouse'];
+    $this->contact_id = $this->createContact($params);
+    // Create Contact
+    $params = ['name' => 'test_case_type', 'title' => 'Test Case Type', 'is_active' => 1];
+    $this->case_type_id = $this->createCaseType($params);
+    // Create Case Status option values
+    $params = ['name' => 'Opened', 'grouping' => 'Opened'];
+    $this->open_status_id = $this->createOptionValue($params, 'case_status');
+  }
+
+  public function tearDown() {
+    parent::tearDown();
+  }
+
+  /**
+   * @dataProvider getUnclosedProvider
+   */
+  public function testGetUnclosedCases($expectedCount, $includeClosed) {
+    $params = [
+      'creator_id' => $this->contact_id,
+      'case_type_id' => $this->case_type_id,
+      'subject' => 'test case',
+      'status_id' => $this->open_status_id,
+    ];
+    $this->createCase($params);
+
+    $params = [
+      'creator_id' => $this->contact_id,
+      'case_type_id' => $this->case_type_id,
+      'subject' => 'test case2',
+      'status_id' => $this->open_status_id,
+      'end_date' => date('Ymd')
+    ];
+    $this->createCase($params);
+
+    $params = [
+      'case_type' => 'Test Case Type',
+    ];
+    $cases = CRM_Tasksassignments_BAO_Assignment::getUnclosedCases($params, [], TRUE, $includeClosed);
+
+    $this->assertCount($expectedCount, $cases);
```

Comment:
This is not a good test. You're only asserting that you have the expected amount, but not the correct cases. If getUnclosedCases returns an array like [1, 2] it will still pass. 

Also, If you decide not to simplify getUnclosedCases, you need to add tests to cover all the possible combinations of the filters (one more reason to simplify it! :) )